aci/resource_aci_fabricnodeidentp.go

```diff
@@ -263,35 +316,112 @@ func resourceAciFabricNodeMemberRead(ctx context.Context, d *schema.ResourceData
 	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
 
 	aciClient := m.(*client.Client)
-
 	dn := d.Id()
 	fabricNodeIdentP, err := getRemoteFabricNodeMember(aciClient, dn)
-
 	if err != nil {
 		return errorForObjectNotFound(err, dn, d)
 	}
+
+	getAndsetDecommissionedNodes(aciClient, d.Get("serial").(string), d)
+
 	_, err = setFabricNodeMemberAttributes(fabricNodeIdentP, d)
 	if err != nil {
 		d.SetId("")
 		return nil
 	}
+
 	log.Printf("[DEBUG] %s: Read finished successfully", d.Id())
 
 	return nil
 }
 
 func resourceAciFabricNodeMemberDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	log.Printf("[DEBUG] %s: Beginning Destroy", d.Id())
-
 	aciClient := m.(*client.Client)
 	dn := d.Id()
-	err := aciClient.DeleteByDn(dn, "fabricNodeIdentP")
+
+	attachedToSwitch, err := verifyNodeAttachedToSwitch(aciClient, d.Get("serial").(string))
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	if attachedToSwitch {
+		decommissionFabricNodeMemberFromController("true", "no", d, m)
+	} else {
+		err = aciClient.DeleteByDn(dn, "fabricNodeIdentP")
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
 
 	log.Printf("[DEBUG] %s: Destroy finished successfully", d.Id())
 
 	d.SetId("")
 	return diag.FromErr(err)
 }
+func verifyNodeAttachedToSwitch(aciClient *client.Client, serial string) (bool, error) {
+	switchStatusDn := fmt.Sprintf("client-[%s]", serial)
```

Comment:
Why not start en end log messages in this function but in the other function there is?

Reply:
see code below where I think it is possible to log the end of this and thus to outcome of the function